documentation/ingestion_server/guides/migrate.md

```diff
@@ -0,0 +1,245 @@
+# Index migration runbook
+
+From time to time, we will need to update our Elasticsearch indices. These
+modifications can be classified into two broad-strokes categories, depending on
+whether the changes affect the main consumer of the indices, the API.
+
+## Migration types
+
+### API-free
+
+These changes are safe modifications to the ES schema that do not affect the
+API. As such they do not need any migration process. Examples:
+
+- addition of new fields or subfields
+- removal of fields that are not referenced or used by the API
+- changing the type to another compatible type (like `text` &harr; `keyword` )
+
+For API-free changes, we deploy the ingestion server and let a data-refresh to
+occur. The indexes will be updated to the new schema without manual intervention
+and will be made available to the API.
+
+### API-involved
+
+These changes are modifications to fields that already are in use by the API and
+involve code changes in both the ingestion server and the API. Examples:
+
+- removal of a field
+- changing the type to an incompatible type
+- renaming of a field
+
+Such kinds of changes need us to precisely deploy the API when the new index is
+promoted because of these reasons:
+
+- If we deploy a little late, the old field the API wants will disappear.
+- If we deploy a little early, the new field the API wants will not be present.
+
+This runbook documents guidelines and processes for API-involved migrations.
+
+Our goal is to break down an API-involved change into multiple small, atomic
+changes with each step affecting at most one of the ingestion server or the API
+and ensuring that the API and ES remain compatible throughout the process.
+
+## Pull request guidelines
+
+A change that involves modification to the ES index as well as its usage in the
+API requires at least three steps, each associated with exactly one PR that
+modifies exactly one of the ingestion server or the API to allow them to be
+deployed independently.
+
+1. Change the ES index mapping in the ingestion server. Ensure that the change
+   is purely additive, keeping the old fields unchanged and creating new fields
+   that contain the data the API will need.
+
+   This PR should only make changes within the `ingestion_server/` directory.
+
+2. Change the ES fields referenced by the API to use the new fields added in the
```

Comment:
This might be a little unclear as "change the ES fields" sounds more like changing something in the Elasticsearch. Maybe something like "Update the API code to reference the new ES field added in the previous step"?

Reply:
Resolved in 128da0b.